File: apps/backend/src/main/java/com/emanagement/backend/modules/attendance/AttendanceService.java
package com.emanagement.backend.modules.attendance;

import com.emanagement.backend.common.dto.PageResponse;
import com.emanagement.backend.modules.attendance.dto.AttendanceHistoryDto;

public interface AttendanceService {
    PageResponse<AttendanceHistoryDto> getUserHistory(Long userId, int page, int size);

    PageResponse<AttendanceHistoryDto> getAllRecords(int page, int size);
}

File: apps/frontend/services/attendance.service.ts
import { apiClient } from "../lib/api-client";
import { ApiResponse, PageResponse } from "../types/common.types";
import { AttendanceHistory } from "../types/attendance.types";

export const attendanceService = {
  getMyHistory: (userId: number, page: number = 0, size: number = 10): Promise<ApiResponse<PageResponse<AttendanceHistory>>> => {
    return apiClient.get<PageResponse<AttendanceHistory>>(`/attendances/my-history?userId=${userId}&page=${page}&size=${size}`);
  },

  getAllRecords: (page: number = 0, size: number = 10): Promise<ApiResponse<PageResponse<AttendanceHistory>>> => {
    return apiClient.get<PageResponse<AttendanceHistory>>(`/attendances/records?page=${page}&size=${size}`);
  },
};

